apis/python/src/tiledbsoma/__init__.py

```diff
@@ -61,15 +61,15 @@
     "TileDBPlatformConfig",
     "TileDBObject",
     "TileDBArray",
-    "SOMACollection",
-    "SOMADataFrame",
-    "SOMADenseNdArray",
-    "SOMADoesNotExistError",
-    "SOMAExperiment",
-    "SOMAError",
-    "SOMAIndexedDataFrame",
-    "SOMAMeasurement",
-    "SOMAMetadataMapping",
-    "SOMASparseNdArray",
+    "Collection",
+    "DataFrame",
+    "DenseNdArray",
+    "DoesNotExistError",
+    "Experiment",
+    "Error",
```

Comment:
`Error` (and its related child classes) may be one place where we want to continue to use `SOMAError` to clearly differentiate from other types of errors/exceptions.

For example, it is `tiledb.TileDBError`, not `tiledb.Error` 

@thetorpedodog - does the Google conventions give any guidance here?

Reply:
Ditto for `DoesNotExistError`

Reply:
One of the recommendations they have is to [not repeat the package name in exceptions as well](https://google.github.io/styleguide/pyguide.html#244-decision):

> Libraries or packages may define their own exceptions. When doing so they must inherit from an existing exception class. Exception names should end in `Error` and should not introduce repetition (`foo.FooError`).

However, skimming through the stdlib documentation I found that `pickle` uses [`pickle.PickleError`](https://docs.python.org/3/library/pickle.html#pickle.PickleError) as a base exception name.

Much like the `SOMAObject` special case, I think `SOMAError` as the base error would be reasonable, but with `DoesNotExistError` as one of the derived types.

Also worth noting is that some of these exception types will no doubt end up moving to the `somabase` package (though moving them there from an initial implementation here seems completely reasonable to me).

Reply:
i.e. `SOMAError` and `DoesNotExist` was my suggestion. `Error` and `DoesNotExist` are also great. I think both are strictly better than `SOMAError` and `SOMADoesNotExist` (which honestly would still be acceptable).

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/464/commits/5a4cac22f860ca9f5fccf89eed834da269be7f7f

Reply:
I can live with any of these.  Consider me still "approved"!  :+1: